### DISCUSSION TITLE
Incorrect carbon intensity rating value returned by the SDK?

### DISCUSSION THREAD

**Discussion by tmcclell** [Green-Software-Foundation/carbon-aware-sdk#141] [General]
For clarification, when calculating SCI, it's usually measured in CO2e and that number usually looks like 951 gCO2e/kWh (example). When I pull emissions data for Azure Region North Central US, it has a rating of 4. What is the unit of measure that the SDK is pulling back? What is "rating" implying?

**Comment by tmcclell** [comment]
Any feedback on this?

**Comment by dtoakley-tw** [comment]
Next steps on this: @tmcclell and @Willmish to take this offline and resolve.

**Comment by Willmish** [comment]
Issue was: appsettings.local.json not being used properly by the SDK, solved by using Environment variables/modifying appsettings.json directly. Fix in PR: https://github.com/Green-Software-Foundation/carbon-aware-sdk/pull/137

**Comment by darren72** [comment]
I would also query @Willmish if the SDK[ ](https://carbon-aware-api.azurewebsites.net/swagger/index.html) is returning the correct values. When I request uksouth data, it returns a value of 449 g/Kwh. When I look up the UK power production values [here](https://carbonintensity.org.uk/) or [here](https://electricityinfo.org/real-time-british-electricity-supply/), they are 1/4 of this. Where did the data actually originate from for the SDK? Is it correct? Am I missing something?

**Reply by Willmish** [reply]
Hi @darren72 , thanks for the question. You can find the data sources supported by the CA SDK here: https://github.com/Green-Software-Foundation/carbon-aware-sdk/blob/dev/docs/selecting-a-data-source.md .

The data provider that is configured in the public Carbon Aware API is WattTime: https://www.watttime.org/

As to why the values are so different as compared to carbonintensity.org.uk and the other source you quoted: This is most likely due to different prediction models being used and possibly different carbon producing elements of the grid being taken into account. As to why exactly the prediction from WattTime differs as much in comparison - I am personally not certain.

 @Henry-WattTime would you mind to weigh in on that topic or point us to some sources?

**Reply by Henry-WattTime** [reply]
Hello @darren72, these sources are measuring different things than each other. The values in the SDK are marginal emissions, or the emissions caused by a change in electricity consumption. This information is useful for taking actions like deciding when or where to run a compute job. The the UK website shows the average emissions factor which includes all resources running at that time. But a nuclear powerplant does not change its output when you consume energy, it will run regardless. The SDK is trying to measure what you can affect with your choices. This paper might help explain this thinking: https://www.watttime.org/app/uploads/2021/08/GHG-Frameworks-WhitePaper-Tomorrow-WattTime-202108.pdf
